src/components/user-auth/accountCreated/AccountCreated.tsx

```diff
@@ -1,29 +1,35 @@
 import { useEffect, useState } from 'react'
 import Logo from '../../logo/Logo'
 import accountConfirmation from '../../../actions/accountConfirmation'
-import { Box, Button, Typography } from '@mui/material'
+import { Box, Button, Typography, Link, CircularProgress } from '@mui/material'
 import { makeStyles } from 'tss-react/mui'
+import { useParams } from 'react-router-dom'
 
 const AccountCreated = () => {
   const { classes } = useStyles()
-  const [success, setSuccess] = useState(false as boolean | undefined)
+  const [success, setSuccess] = useState(false as boolean)
+  const [isLoading, setIsLoading] = useState(true)
+  const { confirmationCode } = useParams()
 
   useEffect(() => {
     async function fetchData() {
-      const { location } = document
-      // It is assumed that the link will look like this http://localhost:3000/account?code=1dfsdfsfd where 1dfsdfsfd is confirmationCode
-      const confirmationCode = location.search.substring(6)
+      // It is assumed that the link will look like this http://localhost:3000/registration/glad-screen/2de95be7-9450-4c3a-80c8-e32584c90315 where 2de95be7-9450-4c3a-80c8-e32584c90315 is confirmationCode
       const result = await accountConfirmation(confirmationCode)
       setSuccess(result)
+      setIsLoading(false)
     }
-    if (success === undefined) {
-      fetchData()
-    }
-  })
+    fetchData()
+  }, [confirmationCode])
 
   return (
     <>
-      {success ? (
+      {isLoading ? (
+        <Box sx={{ display: 'flex', justifyContent: 'center' }}>
+          <CircularProgress
+            style={{ color: '#FB8F67', position: 'absolute', top: '50%' }}
+          />
+        </Box>
+      ) : success ? (
```

Comment:
Bad readability to use nested ternary operators, especially in html part. Plz refactor it

Reply:
Done in last commit